src/js/buffer.js

```diff
@@ -152,7 +152,7 @@ Buffer.prototype.equals = function(otherBuffer) {
     throw new TypeError('Bad arguments: buffer.equals(Buffer)');
   }
 
-  return native.compare(this, otherBuffer) == 0;
+  return this.compare(otherBuffer) == 0;
```

Comment:
Calling an extra function is more costly than using the native function. I don't see much benefit of changing this.